### PR TITLE
fix(twap): display actual value for num of parts and custom deadline

### LINKED ISSUE #2950: Number of parts/total duration is not updated when refresh the page
1. Open Advanced orders page
2. Change a custom duration
3. Change a number of parts
4. Refresh the page

**AR**: number of parts show 2, but 'Sell per part' shows a previous value. 

Custom duration shows a previous value, but it is 0 when open this option
'

**ER**: numbers should be updated and match

### LINKED ISSUE #2951: Don't show TWAP orders when connected to Safe using WC
Discussed in https://cowservices.slack.com/archives/C053B0162VD/p1690482058286619?thread_ts=1690448353.558009&cid=C053B0162VD

Connect to the Safe using WC
Open Advanced orders page
Check Open orders/History tabs

**AR**: there are orders placed in Safe. However, their statuses are not updated appropriately, and it is impossible to cancel orders. 

**ER**: hide orders. Show a message something like "Use the[ Safe web app](https://app.safe.global/share/safe-app?appUrl=https%3A%2F%2Fswap.cow.fi&chain=eth) to see open orders' and ' "Use the[ Safe web app](https://app.safe.global/share/safe-app?appUrl=https%3A%2F%2Fswap.cow.fi&chain=eth) to see orders history'

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2950 #2951

1. There was a problem in `CustomDeadlineSelector`, it set a value for `hoursValue` and `minutesValue` only once
2. In `TradeNumberInput` I've changed the triggers of calling `validateInput` in `useEffect`. I'm afraid it can affect other number inputs. @elena-zh please pay attention to other number inputs on the TWAP form
3. Fixed annoying warnings in `StyledDialogContent`
4. Additionally fixed #2951

image

### CODE REVIEW

File: src/modules/trade/pure/TradeNumberInput/index.tsx [hunk 1]
```diff
@@ -63,8 +63,7 @@ export function TradeNumberInput(props: TradeNumberInputProps) {
   // Initial setup of value
   useEffect(() => {
     validateInput(value ? value.toString() : '')
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
+  }, [value, validateInput])
```
Comment (alfetopito) [inline src/modules/trade/pure/TradeNumberInput/index.tsx:66]: The code looks ok, but this change seems to break the input

I couldn't reproduce on video, but I saw at least twice in the parts input:

1. Focus on parts input
2. Use the arrow keys to move the value up
* Some times - don't know triggered by what - the value jumps back to a lower number of parts

**Review by elena-zh — COMMENTED** [review]
Hey @shoom3301 , great!
However, some nitpicks from my side:
1. When I change a value in the Custon option for Total duration, it is saved without clicking on Apply button
image

2.  I'd write Safe app link in the same manner in both messages 
image
3. And I'd change 'history orders' to 'orders history' 
image

Thanks

File: src/modules/trade/pure/TradeNumberInput/index.tsx (hunk 1 above)
Comment (shoom3301) [reply src/modules/trade/pure/TradeNumberInput/index.tsx]: This is what I exactly was afraid of.
Fixed: https://github.com/cowprotocol/cowswap/pull/2952/commits/9c80334758416e5df335cf637b1214c6f0e8c01f
Thanks!

**Comment by shoom3301** [comment]
@elena-zh @alfetopito thanks! Fixed all

**Review by elena-zh — APPROVED** [review]
LGTM now!

**Review by alfetopito — APPROVED** [review]

**Pull request merged**